source/extensions/common/wasm/context.cc

```diff
@@ -1579,9 +1580,12 @@ Http::FilterHeadersStatus Context::decodeHeaders(Http::RequestHeaderMap& headers
   request_headers_ = &headers;
   end_of_stream_ = end_stream;
   auto result = convertFilterHeadersStatus(onRequestHeaders(headerSize(&headers), end_stream));
-  if (result == Http::FilterHeadersStatus::Continue) {
+  if (http_local_response_sent_) {
+    return Http::FilterHeadersStatus::StopIteration;
```

Comment:
Could you move the override into `convertFilter{Headers,Data,Trailers}Status()` calls instead of doing it inline?